parser/parser.go

```diff
@@ -25,23 +25,39 @@ type (
 var tokEOF = lexer.EOF
 
 var (
-	keywordParsers = map[token.Type]parserfn{}
-	literalParsers = map[token.Type]parserfn{
-		token.Decimal:     parseDecimal,
-		token.Hexadecimal: parseHex,
-		token.String:      parseString,
-		token.Bool:        parseBool,
-		token.Undefined:   parseUndefined,
-		token.Null:        parseNull,
-	}
+	literalParsers map[token.Type]parserfn
 	unaryParsers map[token.Type]parserfn
+	varAssignParsers map[token.Type]parserfn
+	nodeParsers map[token.Type]parserfn
 )
 
 func init() {
 	unaryParsers = map[token.Type]parserfn{
 		token.Minus: parseUnary,
 		token.Plus:  parseUnary,
 	}
+	literalParsers = map[token.Type]parserfn{
```

Comment:
@tiago4orion I moved everything to init for the sake of consistency and also because I wrote a bug before because a variable that could be initialized outside init used the unaryParsers variable but the variable was an empty map at the time since the initialization was not on the order that the variables appeared on the code. If everything was able to be initialized outside init it would be good, but if one needs to be inside init it seems better to all the other ones to be there as well.